### PR TITLE
gcylc dot-view: collapse and expand individual namespaces

### LINKED ISSUE #132: gcylc dot-view: collapse and expand individual namespaces
Currently in the dot-view we can collapse all tasks completely, with member task states shown in hover-over tool-tips, or expand them all completely.  We should probably be able to expand and collapse individual families and tasks, including intermediate namespaces,  as for the other two views.  This is a new feature request though, not a bug, so long as the right-click menu is adapted for the current situation before the next release (#133).

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #132 
Families can now be expanded/collapsed in the dot view.

@hjoliver Please Review
@benfitzpatrick Please Review

### CODE REVIEW

**Comment by hjoliver** [comment]
I've been playing with this with the `examples/task-states` suite, which has one family.  I'm getting quite frequent exceptions like this:
```
/home/vagrant/cylc/cylc.git/lib/cylc/gui/updater_dot.py:296: Warning: Source ID 5244 was not found when attempting to remove it
  self.led_liststore.clear()
```

Also, family-expand sometimes has no effect, and sometimes it quickly collapses again by itself.  @oliver-sanders - do you see these problems?

**Comment by oliver-sanders** [comment]
@hjoliver Fixed:
- Issue where rows collapse on state change
- Issue where clicking on families in transpose mode has no effect

I was unable to replicate your warning message, please say if it persists.

**Comment by hjoliver** [comment]
(as discussed) warning messages still emitted, only for the non-transposed view.

**Comment by oliver-sanders** [comment]
@hjoliver 26209b4e0e2fc57228527eb15703580748ffa407 should suppress these warnings.

File: lib/cylc/gui/updater_dot.py [hunk 1]
```diff
@@ -28,9 +26,16 @@
 from cylc.network.suite_state import get_id_summary
 from copy import deepcopy
 
+import warnings
+warnings.filterwarnings('ignore', 'use the new', Warning)
+
```
Comment (hjoliver) [inline lib/cylc/gui/updater_dot.py]: This isn't working - second arg should match the warning message?

Comment (oliver-sanders) [inline lib/cylc/gui/updater_dot.py]: Derp, have updated the string accordingly.

File: lib/cylc/gui/updater_dot.py [hunk 2]
```diff
@@ -120,25 +127,44 @@ def update(self):
...
+                        self.family_tree[item] = []
+                    self.family_tree[item].append(name)
+
+            for heading, tasks in self.family_tree.iteritems():
+                # Place associated tasks after headers.
+                ind = min([self.task_list.index(task) for task in tasks])
+                for task in tasks:
+                    if task in self.task_list:
+                        self.task_list.remove(task)
+                self.task_list = self.task_list[:ind - 1] + [heading] +\
```
Comment (benfitzpatrick) [inline lib/cylc/gui/updater_dot.py]: For consistency with the rest of the code, it's better to use brackets for continuation lines

**Comment by benfitzpatrick** [comment]
Can we flip the triangle so it points towards the name in non-expanded mode? I think that is the right direction (with my head on its side).

**Comment by oliver-sanders** [comment]
@benfitzpatrick Addressed comments.

**Comment by hjoliver** [comment]
If I start up in the transposed state, the family won't expand.

**Comment by oliver-sanders** [comment]
@hjoliver, @benfitzpatrick 
- Fixed the startup in transpose state bug.
- c8b7fdc52e662c12069020d540934a89e4e82e02 Fixes a bug whereby the transpose view icon in the view's toolbar is not set with the correct state when setting `transpose <dot/graph> = True` in the `gcylc.rc` file.

**Comment by hjoliver** [comment]
That's got it, thanks.  Ben agrees on merge.

**Pull request merged**